Backends/CLX/frame-manager.lisp

```diff
@@ -58,64 +58,28 @@
 
 ;; Abstract pane lookup logic
 
-(defun find-first-defined-class (types)
-  (first
-   (remove-if #'null
-              (mapcar (lambda (class-name)
-                        (find-class class-name nil))
-                      types))))
-
-(defun find-symbol-from-spec (package-spec name-components)
-  (flet ((coerce-name-element (name-elt)
-           (typecase name-elt
-             (symbol (symbol-name name-elt))
-             (sequence (coerce name-elt 'string))
-             (t (princ-to-string name-elt)))))
-  (find-symbol
-   (apply #'concatenate 'string (mapcar #'coerce-name-element name-components))
-   package-spec)))
-
-(defun find-symbols (name-specs)
-  (remove-if #'null (mapcar #'(lambda (x) (find-symbol-from-spec (first x) (rest x))) name-specs)))
-
-(defun generate-standard-pane-specs (type)
-  (let ((mapping (get type 'climi::concrete-pane-class-name)))
-    `((,(symbol-package mapping) ,mapping)
-      (:climi ,mapping)
-      (:climi ,type #:-pane)
-      (:climi ,type))))
-
-(defun generate-clx-pane-specs (type)
-  (append
-   `((:clim-clx #:clx- ,type #:-pane)
-     (:clim-clx #:clx- ,type)
-     (:climi #:clx- ,type #:-pane)
-     (:climi #:clx- ,type))
-   (generate-standard-pane-specs type)))
-
-(defun find-concrete-pane-class (type)
-  (if (or (eql (symbol-package type)
-               (find-package '#:clim))
-          (eql (symbol-package type)
-               (find-package '#:climi))
-          (eql (symbol-package type)
-               (find-package '#:keyword))
-	  (get type 'climi::concrete-pane-class-name))
-      (find-first-defined-class (find-symbols (generate-clx-pane-specs type)))
-      type))
-
-;;; This is an example of how make-pane-1 might create specialized instances of
-;;; the generic pane types based upon the type of the frame-manager. However, in
-;;; the CLX case, we don't expect there to be any CLX specific panes. CLX uses
-;;; the default generic panes instead.
+(defmethod find-concrete-pane-class ((fm clx-frame-manager)
+                                     pane-type &optional (errorp t))
+  ;; This backend doesn't have any specialized pane implementations
+  ;; but depending on circumstances it may add optional mirroring to
+  ;; the class. Such automatically defined concrete class has the same
+  ;; name but with a gensym prefix and symbol in the backend package.
+  (maybe-mirroring fm (call-next-method)))
```

Comment:
I believe this should be &optional errorp … (declare (ignore errorp)).

Reply:
right you are. fixed.